packages/playwright/src/reporters/base.ts

```diff
@@ -254,11 +254,14 @@ export class BaseReporter implements ReporterV2 {
   }
 
   private _printMaxFailuresReached() {
-    if (!this.config.maxFailures)
+    let maxFailures = this.config.maxFailures;
+    maxFailures = Number.isInteger(maxFailures) ? maxFailures : Math.ceil(maxFailures * this.totalTestCount);
```

Comment:
Having this logic here seems fragile - you should probably resolve failures count after parsing the test suite, or explicitly pass a boolean saying that max failures was reached.